sources/api/netdog/src/wicked.rs

```diff
@@ -0,0 +1,504 @@
+        match dhcp4 {
+            Dhcp4Config::DhcpEnabled(b) => WickedDhcp4 {
+                enabled: Some(b),
+                _f: (),
+                ..Default::default()
+            },
+            Dhcp4Config::WithOptions(o) => WickedDhcp4 {
+                enabled: o.enabled,
+                route_priority: o.route_metric,
+                _f: (),
+                ..Default::default()
+            },
+        }
+    }
+}
+
+impl WickedInterface {
+    pub(crate) fn name(&self) -> &str {
+        &self.name
+    }
+
+    /// Create a WickedInterface given a name and configuration
+    pub(crate) fn from_config(name: InterfaceName, config: NetInterface) -> Self {
+        let wicked_dhcp4 = config.dhcp4.map(WickedDhcp4::from);
+        // As additional options are added for IPV6, implement `From` similar to WickedDhcp4
+        let wicked_dhcp6 = config.dhcp6.map(|dhcp6_enabled| WickedDhcp6 {
+            enabled: Some(dhcp6_enabled),
+            defer_timeout: None,
+            flags: None,
+            _f: (),
+        });
+        WickedInterface {
+            name,
+            control: WickedControl::default(),
+            ipv4_dhcp: wicked_dhcp4,
+            ipv6_dhcp: wicked_dhcp6,
+        }
+    }
+
+    /// Serialize the interface's configuration file
+    // Consume `self` to enforce that changes aren't made to the interface type after it has been
+    // written to file
+    pub(crate) fn write_config_file(&self) -> Result<()> {
+        let mut cfg_path = Path::new(WICKED_CONFIG_DIR).join(self.name.to_string());
+        cfg_path.set_extension(WICKED_FILE_EXT);
+
+        // TODO: pretty print these files.  `serde_xml_rs` doesn't support pretty printing;
+        // `quick_xml` does, however we require a few features that haven't been released yet to
+        // properly serialize the above data structures:
+        // https://github.com/tafia/quick-xml/issues/340#issuecomment-981093602
+        let xml = serde_xml_rs::to_string(&self).context(error::XmlSerializeSnafu {
+            interface: self.name.to_string(),
+        })?;
+        fs::write(&cfg_path, xml).context(error::WickedConfigWriteSnafu { path: cfg_path })
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// Allow a simple interface definition to be parsed from a string.  The expected input string
+/// looks like: `interface-name:option1,option2`.  The colon is required.  Acceptable options are
+/// "dhcp4", and "dhcp6".  For both options an additional sigil, "?", may be provided to signify
+/// that the protocol is optional.  "Optional" in this context means that we will not wait for a
+/// lease in order to consider the interface operational.
+///
+/// An full and sensible example could look like: `eno1:dhcp4,dhcp6`.  This would create an
+/// interface configuration for the interface named `eno1`, enable both dhcp4 and dhcp6, and
+/// consider a dhcp6 lease optional.
```

Comment:
Wouldn't DHCP6 be required in the case of `eno1:dhcp4,dhcp6`?

Reply:
Yep- this was a typo. Fixed